cupy/core/_routines_math.pyx

```diff
@@ -213,35 +368,117 @@ cdef ndarray scan(ndarray a, ndarray out=None):
 
     """
     if a._shape.size() != 1:
-        raise TypeError('Input array should be 1D array.')
+        a = a.ravel()
 
     cdef Py_ssize_t block_size = 256
-
     if out is None:
-        out = ndarray(a.shape, dtype=a.dtype)
+        out = ndarray(a.shape, a.dtype)
     else:
         if a.size != out.size:
             raise ValueError('Provided out is the wrong size')
 
-    kern_scan = _inclusive_scan_kernel(a.dtype, block_size)
+    kern_scan = _inclusive_scan_kernel(a.dtype, block_size, op)
     kern_scan(grid=((a.size - 1) // (2 * block_size) + 1,),
               block=(block_size,),
               args=(a, out),
               shared_mem=a.itemsize * block_size * 2)
 
     if (a.size - 1) // (block_size * 2) > 0:
         blocked_sum = out[block_size * 2 - 1:None:block_size * 2]
-        scan(blocked_sum, blocked_sum)
-        kern_add = _add_scan_blocked_sum_kernel(out.dtype)
+        _scan_op(blocked_sum, op, dtype, blocked_sum)
+        kern_add = _add_scan_blocked_sum_kernel(out.dtype, op)
         kern_add(grid=((a.size - 1) // (2 * block_size),),
                  block=(2 * block_size - 1,),
                  args=(out,))
     return out
 
 
+cdef ndarray _batch_scan_op(ndarray a, op, dtype=None, out=None):
+    batch_size = a.shape[1]
+
+    block_size = 512
```

Comment:
```suggestion
    block_size = _reduction._block_size
```

Reply:
How is this comment?

Reply:
This symbol was not accessible from a cdef as it is not declared in the `pxd` file, 
If I move it than lots of unrelated errors happens 😢,
Maybe it should be fixed in another PR?

Reply:
OK. Could you leave a comment?

Reply:
Done! thanks